node/node.go

```diff
@@ -14,7 +15,7 @@ import (
 )
 
 type StarknetNode interface {
-	Run() error
+	Run(ctx context.Context) error
```

Comment:
We have an explicit `Shutdown()`, so we don't need a `ctx` here. Alternative is getting rid of `Shutdown()`. Same goes for Synchronizer

Reply:
I think getting rid of Shutdown is the better option here btw

Reply:
That's a great point! Didn't think of that.